### PR TITLE
Use Test Scope Configurations in Identity

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Removes all mutations of global configuration in `azure-identity` tests and replaces it with `Configuration`s with set-up scoped to the specific test.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/identity/azure-identity/src/test/java/com/azure/identity/util/TestUtils.java [hunk 1]
```diff
@@ -135,6 +141,17 @@ public static Mono<AccessToken> getMockAccessToken(String accessToken, OffsetDat
...
     }
 
+    /**
+     * Creates a {@link Configuration} with the specified {@link ConfigurationSource} as the only source of
+     * configurations.
+     *
+     * @param configurationSource The configuration source.
+     * @return A configuration used for testing.
+     */
+    public static Configuration createTestConfiguration(ConfigurationSource configurationSource) {
```
Comment (billwert) [inline sdk/identity/azure-identity/src/test/java/com/azure/identity/util/TestUtils.java:151]: nit: Can this optionally take a `null` and create a default `TestConfigurationSource` if so? All the callsites having to create one isn't great.

File: sdk/identity/azure-identity/src/test/java/com/azure/identity/implementation/IdentityClientTests.java [hunk 2]
```diff
@@ -222,99 +260,111 @@ public void testValidIdentityEndpointMSICodeFlow() throws Exception {
             .setManagedIdentityType(ManagedIdentityType.APP_SERVICE)
             .setManagedIdentityParameters(new ManagedIdentityParameters()
                 .setIdentityEndpoint(endpoint)
-                .setIdentityHeader(secret));
+                .setIdentityHeader(secret))
+            .setConfiguration(configuration);
```
Comment (billwert) [inline sdk/identity/azure-identity/src/test/java/com/azure/identity/implementation/IdentityClientTests.java:264]: indentation (and the next few too)

**Review by billwert — APPROVED** [review]
couple minor comments but looks great

File: sdk/identity/azure-identity/src/test/java/com/azure/identity/implementation/IdentityClientTests.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/identity/azure-identity/src/test/java/com/azure/identity/implementation/IdentityClientTests.java:264]: This is the correct indentation as the deeper indented lines are creating an Object which closes before setting configuration

Comment (billwert) [reply sdk/identity/azure-identity/src/test/java/com/azure/identity/implementation/IdentityClientTests.java:264]: Oh I missed the second constructor, my bad.

**Pull request merged**